### PR TITLE
apps: Upgrade kube-prometheus-stack to v49.2.0

### LINKED ISSUE #1486: [3] Upgrade Kube-prometheus-stack
**What Chart should be upgraded?**

upstream/Kube-prometheus-stack chart to the latest stable version. https://github.com/prometheus-community/helm-charts/releases

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by Pavan-Gunda**
### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Critical security fixes should be marked with `kind/security`
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

...

- Fixes #1486

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [x] Any changed pod is covered by Pod Security Admission
  - [x] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [x] The change does not cause any alerts to be generated by Falco

### CODE REVIEW

File: migration/v0.33/apply/11-prometheus-operator-crds.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+run() {
+  case "${1:-}" in
+  execute)
+    for CLUSTER in sc wc; do
+      log_info "  - applying the prometheus-operator CRDs on $CLUSTER"
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml
```
Comment (Xartos) [inline migration/v0.33/apply/11-prometheus-operator-crds.sh:13]: Could you change the URLs to point to the files in the upstream folder instead? Since we already have them we should remove the dependency to having github access

File: migration/v0.33/apply/11-prometheus-operator-crds.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_probes.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.67.1/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
+    done
+    ;;
+  rollback)
+    log_warn "rollback not implemented"
```
Comment (Xartos) [inline migration/v0.33/apply/11-prometheus-operator-crds.sh:24]: Can you here use the `apply set-last-applied` command to revert the change?

File: migration/v0.33/apply/11-prometheus-operator-crds.sh (hunk 1 above)
Comment (Pavan-Gunda) [reply migration/v0.33/apply/11-prometheus-operator-crds.sh:13]: Sure !!

File: migration/v0.33/README.md [hunk 3]
```diff
@@ -193,6 +193,12 @@ As with all scripts in this repository `CK8S_CONFIG_PATH` is expected to be set.
     ./migration/v0.33/apply/40-rclone.sh execute
     ```
 
+1. Apply the new kube-prometheus-stack CRDs:
+
+    ```bash
+    ./migration/v0.30/apply/11-prometheus-operator-crds.sh execute
```
Comment (crssnd) [inline migration/v0.33/README.md:199]: will this be part of apps 0.33 or 0.34?

File: WIP-CHANGELOG.md [hunk 4]
```diff
@@ -30,6 +30,7 @@
 
 - Upgraded falco-exporter chart version to `v0.9.6` and app version to `v0.8.3`
 - Added "IPs in Subnet mask" logic in checkIfDiffAndUpdateDNSIPs, checkIfDiffAndUpdateKubectlIPs and checkIfDiffAndUpdateIPs of update-ips.bash
+- Upgrade kube-prometheus-stack chart version from `v45.2.0` to `v49.2.0` and app version from `v0.63.0` to `v0.67.1`
```
Comment (crssnd) [inline WIP-CHANGELOG.md:33]: any user facing changes that we and they should be aware?

**Comment by crssnd** [comment]
can you also fill in the PR form? maybe use the new one

File: migration/v0.33/README.md (hunk 3 above)
Comment (Pavan-Gunda) [reply migration/v0.33/README.md:199]: I moved the migration doc to v34

File: WIP-CHANGELOG.md (hunk 4 above)
Comment (Pavan-Gunda) [reply WIP-CHANGELOG.md:33]: None as far as i'm aware !!

**Review by Xartos — APPROVED** [review]

File: migration/v0.34/apply/11-prometheus-operator-crds.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-podmonitors.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-probes.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheuses.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheusrules.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml
+      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-thanosrulers.yamlset-last-applied
```
Comment (crssnd) [inline migration/v0.34/apply/11-prometheus-operator-crds.sh:20]:
```suggestion
      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-thanosrulers.yaml
```

**Comment by crssnd** [comment]
after the upgrade prometheus cannot scrape the alert manager config reloader due to networkpolicy
can you re-check that you do not have packets dropped by netpol?
can you also check that all targets are up after the upgrade?

**Review by crssnd — APPROVED** [review]
do not forget to squash your commits and finish filling in the PR form
